### PR TITLE
ELI5, Download pushshift Reddit and CC Dumps

### PR DESCRIPTION
**Pull request by domrigoglioso**
**Patch description**
Adding the ELI5 dataset and capability for downloading specific Reddit and Common Crawl dumps  as in #2094. There's an overview of changes for each ELI5 repo file in the other information section.
 
**Testing steps**
Testing `build.py`:
Run `python examples/display_data.py -t eli5` and checked directory structure and pre-computed files. `eli5` data folder had `eli5/pre_computed`, `eli5/processed_data`, `eli5/tmp`,
`eli5/processed_data/collected_docs` directories as expected. `wet.paths`, `explainlikeimfive_ccrawls_ids.json`, and `explainlikeimfive_unigram_counts` were also downloaded in `pre_computed` as expected.

Testing `agents.py`:
Took https://github.com/facebookresearch/ELI5/blob/master/data_creation/processed_data/public_examples/examples_qda_short.json since I can't run all the steps, and saved it as `eli5/processed_data/selected_15_1/explainlikeimfive_train.json`.
Ran `python examples/display_data.py -t eli5`,  single episode here:
```
6 . A HERBIVORE is a living thing that eats only plants . One example of this is a DEER . [... I removed a bunch of lines to make this smaller ...] Herbivores are also known as herbivorous animals .
If an animal is an herbivore ( only eats plants ) , where does it get the protein that it needs to survive if it only eats grass ? --T-- 
   Herbivores only eat plants , but unlike humans they have special digestive systems that let them get protein from grass . Plants contain cellulose , which is difficult for humans to digest . However , animals like cows have special digestive systems that can process the cellulose and then access nutrition like protein that are protected by the cellulose in plant cells . A lot of herbivores have multiple stomachs ( like cows ) or special areas in the stomach or intestine that can help in digesting plants .
```

Ran `python examples/display_data.py --knowledge False -t eli5`, single episode here:
```
If an animal is an herbivore ( only eats plants ) , where does it get the protein that it needs to survive if it only eats grass ? --T-- 
   Herbivores only eat plants , but unlike humans they have special digestive systems that let them get protein from grass . Plants contain cellulose , which is difficult for humans to digest . However , animals like cows have special digestive systems that can process the cellulose and then access nutrition like protein that are protected by the cellulose in plant cells . A lot of herbivores have multiple stomachs ( like cows ) or special areas in the stomach or intestine that can help in digesting plants .
```

To test downloading Reddit post ids:
`python3 download_reddit_qalist.py -id_l posts.json -Q`
`python3 download_reddit_qalist.py -id_l posts.json -A`
and checked the jsons in `processed_data/` had the posts and comments in them.
`posts.json` is a .json file containing the following:  `["fhm9mp", "fhpcke"]`
Also ran `python3 download_reddit_qalist.py -Q -sy 2011 -ey 2011` and `python3 download_reddit_qalist.py -A -sy 2011 -ey 2011` to make sure those still worked.

Testing `download_support_docs.py`:
Edited `wet.paths` to only contain the following:
`crawl-data/CC-MAIN-2018-34/segments/1534221208676.20/wet/CC-MAIN-20180814062251-20180814082251-00000.warc.wet.gz`

Ran `python3 parlai/tasks/eli5/download_support_docs.py -u urls.json`
where `urls.json` contained the text ` ["https://openeffect.com/"]`
and checked that the `slices/0/`'s 0000 slice contained the openeffect url, id, and text.

Ran `python3 parlai/tasks/eli5/download_support_docs.py -ids ids.json`
where `ids.json` contained the text ` ["aec9fe35-7fec-46b3-96de-dd2db8887902"]` or alternatively `["<urn:uuid:aec9fe35-7fec-46b3-96de-dd2db8887902>"]` (this is the CCID corresponding with openeffect.com)
and checked that the `slices/0/`'s 0000 slice contained the openeffect url, id, and text.
Slice content for both for reference:
```
[[0, {"ccid": "<urn:uuid:aec9fe35-7fec-46b3-96de-dd2db8887902>", "url": "http://openeffect.com/", "text": ["This site requires JavaScript and Cookies to be enabled . Please change your browser settings or upgrade your browser .", []]}]]
```
Testing `merge_support_docs.py`:
Ran `python3 parlai/tasks/eli5/merge_support_docs.py -c 0 -n specific_ids` and
`python3 parlai/tasks/eli5/merge_support_docs.py -c finalize -n specific_ids`
Checked `0.json` and `slices/slice_0.json` had same text as above slice.

**Other information**
Overview of Changes from github.com/facebookresearch/ELI5/:
`download_support_docs.py`: Changed to use ParlAI parser and datapath, added ability to download specific urls and ids from Common Crawl
Added `select_specific_ids`, `select_specific_urls`, `check_url`, code to differentiate between downloading specific posts, ccids, or eli5 docs. The `main` code for downloading and saving docs was from ELI5.

`download_reddit_qalist.py`: Changed to use ParlAI parser and datapath, added ability to download specific reddit post ids
Added `download_and_process_comments`, `download_and_process_posts`, `get_posts`, `get_comments_from_post`, and a few parts of the parsing part main for downloading specific reddit posts. Rest is from ELI5.

`merge_support_docs.py`: Changed to use ParlAI parser and datapath. Added `merge_non_eli_docs` and a few lines in `main` to merge non-eli5 docs.

`data_creation/data_utils.py`: No changes except lint
`data_creation/finalize_qda.py`,  `/select_sentences_tfidf.py`: Changed to use ParlAI parser and datapath
`data_creation/slurm_scripts/eli_download_docs.sh`, `/eli_merge_docs.sh`: Changed to use relative `data_creation` directory rather than hard-coded directory
`data_creation/slurm_scripts/eli_select_docs.sh`: Changed to use relative `data_creation` directory rather than hard-coded directory, changed to use parser arguments instead of sysv args (due to `select-sentences_tfidf.py` changes)
All other files in `slurm_scripts/` were unchanged
`README.md`: Changes to mention the ELI5 repo, mention automatic downloading of pre-computed files, section at end for general reddit downloading and common crawl ids/urls.

**Data tests (if applicable)**
If you added a new teacher, you will be asked to run
`python setup.py test -s tests.suites.datatests`. Please paste this log here.

I don't have the resources to download everything and run the slurm scripts, so I can't confirm the final dataset.

### CODE REVIEW

**Comment by stephenroller** [comment]
I'm so excited you're doing this!

I'm very pleased with the example jsons. Such examples lend a lot of clarity. I'm going to check some facts around just how many examples we feel comfortable reproducing in this repo.

**Comment by domrigoglioso** [comment]
@stephenroller You mentioned in the original issue about making the code for the eli5 teacher public. Would I be able to get that to use in the `agents.py` for this?

File: parlai/tasks/eli5/README.md [hunk 1]
```diff
@@ -0,0 +1,143 @@
...
+
+--------------------------------------------------------------------------------
+
+# Data creation
+
+We provide a suite of scripts to download paired questions and answers from the ELI5 subreddit along with supporting documents from the CommonCrawl
+
+### FAQ: can you provide the processed data?
+
+No, we are not allowed to host processed Reddit or CommonCrawl data. While we are aware that it would make life a little easier, making your own version of the dataset following the instructions here is pretty straightforward if you have access to a SLURM cluster. **We're also happy to work with you if your cluster works on another operating system, open an issue and we'll get on it!** Since the creation process involves downloading, filtering and tokenizing a full CommonCrawl dump however, using a single machine is impractical.
```
Comment (klshuster) [inline parlai/tasks/eli5/README.md:21]: we should get rid of this FAQ - "we" is referring to the ELI5 repo owners, but we as ParlAI are not speaking on their behalf. If anything, please direct people to the ELI5 repository for help with cluster stuff etc.

File: parlai/tasks/eli5/README.md [hunk 2]
```diff
@@ -0,0 +1,143 @@
+# ELI5
+**Adapted from Facebook Research's ELI5 found at https://github.com/facebookresearch/ELI5/**
```
Comment (klshuster) [inline parlai/tasks/eli5/README.md:2]: Thanks for including a README. In general, I think it would be better if a few parts of this were rewritten from the ParlAI standpoint, i.e. as a platform that provides these scripts as a way to download and use ELI5 _within ParlAI_; as such, a lot of the talking points here should be left in the regular ELI5 README and perhaps just  _referenced_ from this README. I won't correct everything I see but will make comments here and there and I think you'll get the general idea of what we want this README to look like

File: parlai/tasks/eli5/README.md [hunk 3]
```diff
@@ -0,0 +1,143 @@
...
+## Overview of the data creation process
+
+The process consists of four steps. *Steps 1 and 2 should be run in parallel.*
+
+1. **Run build.py to setup the directory and download pre-computed files**. This only takes a few minutes.
+2. **Downloading and filtering the Reddit data.** This can be run on a single machine and may take up to 72 hours.
+3. **Downloading and tokenizing the CommonCrawl pages.** This part requires access to a cluster. We provide a sample SLURM script using 100 threads, which on our cluster finishes in under 48 hours.
+4. **Selecting passages from the downloaded pages to create the final support document.** After running steps 1 and 2, this part uses our TFIDF heuristic to create the final ~1000 words support document, and create a train/valid/test split of Question-Document-Answer triplets.
+
+If you are having trouble with any of these, please open an issue stating which step is failing in the title, and attach the Python error text, if available.
```
Comment (klshuster) [inline parlai/tasks/eli5/README.md:33]: e.g. here, we should have people open the issue in ELI5 i think

File: parlai/tasks/eli5/README.md [hunk 4]
```diff
@@ -0,0 +1,143 @@
...
+The first line takes about 6 hours on one machine to download the questions, and the second less than 48 hours for the answers. Pushshift files are automatically removed after they've been processed, so space shouldn't be an issue there. The final product should be 689MB.
+
+
+## Downloading support documents from the CommonCrawl
+
+We provide a list of CommonCrawl IDs for supporting documents for each of the questions.
+These are obtained from `build.py` which we went over above.
+
+
+The next step than consists in reading through the CommonCrawl WET files to gather the text of pages which are used as support documents. In order to gather the documents for each QA pair (up to 100 per pair, sometimes less after deduplicating) using a SLURM cluster and 100 threads, run:
```
Comment (klshuster) [inline parlai/tasks/eli5/README.md:58]: nit: "the next step *then*"

File: parlai/tasks/eli5/README.md [hunk 5]
```diff
@@ -0,0 +1,143 @@
...
+```
+
+## download_support_docs.py
+The following download Common Crawl text of URLs contained in [url_list.json] that are in Common Crawl dumps in `wet.paths`. [url_list.json] should be .json file of a list of URLs.
+```
+python download_support_docs.py -u [url_list.json]
+python download_support_docs.py --urls [url_list.json]
+```
+
+The following download Common Crawl text of URLs contained in [id_list.json] that are in Common Crawl dumps in `wet.paths`. [id_list.json] should be .json file of a list of URLs.
```
Comment (klshuster) [inline parlai/tasks/eli5/README.md:136]: nit: is this a list of URLs or IDs?

File: parlai/tasks/eli5/agents.py [hunk 6]
```diff
@@ -0,0 +1,74 @@
...
+
+
+class ELI5Teacher(FixedDialogTeacher):
+    """
+    ELI5 Teacher, taken from https://github.com/facebookresearch/ELI5.
+    """
+
+    def __init__(self, opt, shared=None):
+        super().__init__(opt, shared)
+        # self.add_cmdline_args(opt)
```
Comment (klshuster) [inline parlai/tasks/eli5/agents.py:20]: nit: remove commented code

File: parlai/tasks/eli5/agents.py [hunk 7]
```diff
@@ -0,0 +1,74 @@
...
+        build(opt)
+        # self._setup_data(opt['datafile'], jsons_path)
```
Comment (klshuster) [inline parlai/tasks/eli5/agents.py:22]: ditto

File: parlai/tasks/eli5/agents.py [hunk 8]
```diff
@@ -0,0 +1,74 @@
...
+        dp = opt['datapath']
+        dt = opt['datatype'].split(':')[0]
+        eli_path = "eli5/processed_data/selected_15_1/explainlikeimfive_"
+        fname = os.path.join(dp, eli_path + dt + ".json")
+        opt['datafile'] = fname
+        with open(fname) as json_file:
+            data = json.load(json_file)
+        ds = []
+        for d in data:
+            if bool(self.opt['knowledge']) and self.opt['knowledge'] != 'False':
```
Comment (klshuster) [inline parlai/tasks/eli5/agents.py:50]: nit: you don't need the `bool()` check here, having an arg type of `'bool'` auto-parses `False,false,0` and `True,true,1` to be `False` and `True` respectively

File: parlai/tasks/eli5/build.py [hunk 9]
```diff
@@ -0,0 +1,113 @@
...
+
+# Copyright (c) Facebook, Inc. and its affiliates.
+# This source code is licensed under the MIT license found in the
+# LICENSE file in the root directory of this source tree.
+# Download and build the data if it does not exist.
+
+from parlai.core.build_data import DownloadableFile
+import parlai.core.build_data as build_data
+import subprocess
+import os
```
Comment (klshuster) [inline parlai/tasks/eli5/build.py:11]: nit: please remove `os`

File: parlai/tasks/eli5/build.py [hunk 10]
```diff
@@ -0,0 +1,113 @@
...
+
+def build(opt):
+    dpath = pjoin(opt['datapath'], 'eli5')
+    pre_computed_path = pjoin(dpath, 'pre_computed')
+    processed_path = pjoin(dpath, 'processed_data')
+
+    version = '1.0'
+
+    if not build_data.built(dpath, version_string=version):
+        print('[building image data: ' + dpath + ']')
```
Comment (klshuster) [inline parlai/tasks/eli5/build.py:57]: nit: not building "image" data

File: parlai/tasks/eli5/build.py [hunk 11]
```diff
@@ -0,0 +1,113 @@
...
+                f'Unzip wet.paths.gz located at {pre_computed_path} with command:\n',
+                'gunzip wet.paths.gz',
+            )
+        if not isfile(pjoin(pre_computed_path, 'explainlikeimfive_ccrawl_ids.json')):
+            print(
+                f'Unzip explainlikeimfive_ccrawl_ids.json.gz located at {pre_computed_path} with command:\n',
+                'gunzip explainlikeimfive_ccrawl_ids.json.gz',
+            )
+        # Mark the data as built.
+        build_data.mark_done(dpath, version_string=version)
```
Comment (klshuster) [inline parlai/tasks/eli5/build.py:99]: do we want to mark as `built` even if the files are not unzipped accordingly? we don't need to redownload each call to `build` but maybe we should print these instructions each time `build.py` is called?

File: parlai/tasks/eli5/build.py [hunk 12]
```diff
@@ -0,0 +1,113 @@
...
+
+    # Check that reddit file is there for eli5
+    sel_path = pjoin(processed_path, 'selected_15_1')
```
Comment (klshuster) [inline parlai/tasks/eli5/build.py:102]: what is `selected_15_1`?

File: parlai/tasks/eli5/data_creation/data_utils.py [hunk 13]
```diff
@@ -0,0 +1,224 @@
...
+import math
+import re
+import signal
+import subprocess
+
+from contextlib import contextmanager
+from glob import glob
+from os.path import join as pjoin
+from os.path import isfile
+from os.path import isdir
```
Comment (klshuster) [inline parlai/tasks/eli5/data_creation/data_utils.py:22]: nit: please fix lint errors

File: parlai/tasks/eli5/data_creation/data_utils.py [hunk 14]
```diff
@@ -0,0 +1,224 @@
...
+
+from spacy.lang.en import English
```
Comment (klshuster) [inline parlai/tasks/eli5/data_creation/data_utils.py:24]: do we need to add `spacy` to `requirements.txt` at the top level?

File: parlai/tasks/eli5/data_creation/data_utils.py [hunk 15]
```diff
@@ -0,0 +1,224 @@
+#!/usr/bin/env python3
+
+# Copyright (c) Facebook, Inc. and its affiliates.
+# This source code is licensed under the MIT license found in the
+# LICENSE file in the root directory of this source tree.
+
+"""
+Taken from
```
Comment (klshuster) [inline parlai/tasks/eli5/data_creation/data_utils.py:8]: is this a direct copy?

File: parlai/tasks/eli5/data_creation/download_reddit_qalist.py [hunk 16]
```diff
@@ -0,0 +1,539 @@
+#!/usr/bin/env python3
+
+# Copyright (c) Facebook, Inc. and its affiliates.
+# This source code is licensed under the MIT license found in the
+# LICENSE file in the root directory of this source tree.
+
+"""
+Adapted from https://github.com/facebookresearch/ELI5/blob/master/data_creation/download
```
Comment (klshuster) [inline parlai/tasks/eli5/data_creation/download_reddit_qalist.py:8]: how much of this is copied?

File: parlai/tasks/eli5/data_creation/slurm_scripts/eli_download_docs.sbatch [hunk 17]
```diff
@@ -0,0 +1,10 @@
+#!/bin/bash
+# Copyright (c) Facebook, Inc. and its affiliates. All Rights Reserved
```
Comment (klshuster) [inline parlai/tasks/eli5/data_creation/slurm_scripts/eli_download_docs.sbatch:2]: nit: should filename be `eli5_download_docs.sh` instead?

**Comment by domrigoglioso** [comment]
@klshuster A lot of the files rely on a hardcoded directory, so would it be best to just tell them to clone the ELI5 repo and run the ELI5 creation steps there and just have a script that moves the data over from their ELI5 folder to their ParlAI data folder (or just tell them to move the final data over)? If we did do that, we'd still need to keep `download_reddit_qalist.py`, `download_support_docs.py`, `merge_support_docs.py`, and some of `data_utils.py`, to download miscellaneous reddit posts and CC docs. For that would I just move any necessary functions from `data_utils.py` into whatever files they get used in for those 3 files above?

File: parlai/tasks/eli5/build.py (hunk 12 above)
Comment (domrigoglioso) [reply parlai/tasks/eli5/build.py:102]: This is just the directory that gets made when you run the last script (`finalize_qda.py`) which holds the test, train, valid.

**Comment by klshuster** [comment]
After re-reading the original issue (#2094), I'll walk back some of my suggestions. Indeed it might be useful to simply host the scripts in ParlAI, under the assumption that they'll make accessing the dataset an easier experience.

With that being said, I have a couple more asks before we can merge this:

1. Let's clean up the lint errors a bit
2. Could we document some of the individual files just a bit more to add clarity
3. Finally, could you indicate in the PR which parts of the files are direct copies from ELI5 and which parts are adapted to ParlAI usage?

File: parlai/tasks/eli5/data_creation/download_reddit_qalist.py (hunk 16 above)
Comment (domrigoglioso) [reply parlai/tasks/eli5/data_creation/download_reddit_qalist.py:8]: I added download_and_process_comments, download_and_process_posts, get_posts, get_comments_from_post, and a few parts of the parsing part main for downloading specific reddit posts. Rest is from ELI5.

File: parlai/tasks/eli5/data_creation/data_utils.py (hunk 15 above)
Comment (domrigoglioso) [reply parlai/tasks/eli5/data_creation/data_utils.py:8]: Yup (except lint changes)

**Comment by klshuster** [comment]
Thanks for addressing my changes, maybe @stephenroller could give a _quick_ glance to make sure I'm not missing anything basic (i.e. whether we'd need to include ELI5 license etc.)

File: parlai/tasks/eli5/data_creation/data_utils.py (hunk 14 above)
Comment (stephenroller) [reply parlai/tasks/eli5/data_creation/data_utils.py:21]: Not at the top level. We should wrap it in a try/except here saying to go to spacy.io and follow installation instructions.

File: parlai/tasks/eli5/data_creation/data_utils.py [hunk 18]
```diff
@@ -0,0 +1,214 @@
...
...|sv|sx|sy|sz|tc|td|tf|tg|th|tj|tk|tl|tm|tn|to|tp|tr|tt|tv|tw|tz|ua|ug|uk|us|uy|uz|va|vc|ve|vg|vi|vn|vu|wf|ws|ye|yt|yu|za|zm|zw)/)(?:[^\s()<>{}\[\]]+|\([^\s()]*?\([^\s()]+\)[^\s()]*?\)|\([^\s]+?\))+(?:\([^\s()]*?\([^\s()]+\)[^\s()]*?\)|\([^\s]+?\)|[^\s`!()\[\]{};:'".,<>?«»“”‘’])|(?:(?<!@)[a-z0-9]+(?:[.\-][a-z0-9]+)*[.](?:com|net|org|edu|gov|mil|aero|asia|biz|cat|coop|info|int|jobs|mobi|museum|name|post|pro|tel|travel|xxx|ac|ad|ae|af|ag|ai|al|am|an|ao|aq|ar|as|at|au|aw|ax|az|ba|bb|bd|be|bf|bg|bh|bi|bj|bm|bn|bo|br|bs|bt|bv|bw|by|bz|ca|cc|cd|cf|cg|ch|ci|ck|cl|cm|cn|co|cr|cs|cu|cv|cx|cy|cz|dd|de|dj|dk|dm|do|dz|ec|ee|eg|eh|er|es|et|eu|fi|fj|fk|fm|fo|fr|ga|gb|gd|ge|gf|gg|gh|gi|gl|gm|gn|gp|gq|gr|gs|gt|gu|gw|gy|hk|hm|hn|hr|ht|hu|id|ie|il|im|in|io|iq|ir|is|it|je|jm|jo|jp|ke|kg|kh|ki|km|kn|kp|kr|kw|ky|kz|la|lb|lc|li|lk|lr|ls|lt|lu|lv|ly|ma|mc|md|me|mg|mh|mk|ml|mm|mn|mo|mp|mq|mr|ms|mt|mu|mv|mw|mx|my|mz|na|nc|ne|nf|ng|ni|nl|no|np|nr|nu|nz|om|pa|pe|pf|pg|ph|pk|pl|pm|pn|pr|ps|pt|pw|py|qa|re|ro|rs|ru|rw|sa|sb|sc|sd|se|sg|sh|si|sj|Ja|sk|sl|sm|sn|so|sr|ss|st|su|sv|sx|sy|sz|tc|td|tf|tg|th|tj|tk|tl|tm|tn|to|tp|tr|tt|tv|tw|tz|ua|ug|uk|us|uy|uz|va|vc|ve|vg|vi|vn|vu|wf|ws|ye|yt|yu|za|zm|zw)\b/?(?!@)))"""
```
Comment (stephenroller) [inline parlai/tasks/eli5/data_creation/data_utils.py:42]: break this into longer lines. python supports implict string concatenation.

```python
URL_REGEX = (
    r"(?i)\b((?:https?:(?:/{1,3}|[a-z0-9%])|[a-z0-9.\-]+[.]"
    r"(?:com|net|org|edu|gov|mil|aero|asia|biz|cat|coop|info|int|jobs|mobi|"
    r"museum|name|post|pro|tel|travel|xxx|ac|ad|ae|af|ag|ai|al|am|an|ao"
)
```
and so on

File: parlai/tasks/eli5/data_creation/select_sentences_tfidf.py [hunk 19]
```diff
@@ -0,0 +1,160 @@
...
+        if not isdir(selected_dir):
+            os.mkdir(selected_dir)
+        reddit_dir = pjoin(
+            eli_path, 'processed_data/selected_%d_%d/%s' % (n_sents, n_context, reddit)
+        )
+        if not isdir(reddit_dir):
+            os.mkdir(reddit_dir)
+        selected_slice_json = pjoin(
+            eli_path,
+            'processed_data/selected_%d_%d/%s/selected_slice_%d.json'
```
Comment (stephenroller) [inline parlai/tasks/eli5/data_creation/select_sentences_tfidf.py:153]: nit: f-strings are good options, but this is also fine.

File: parlai/tasks/eli5/data_creation/slurm_scripts/eli_download_docs.sbatch (hunk 17 above)
Comment (stephenroller) [reply parlai/tasks/eli5/data_creation/slurm_scripts/eli_download_docs.sbatch:2]: I like sbatch, it emphasizes that this is best on slurm.

File: parlai/tasks/eli5/README.md (hunk 2 above)
Comment (stephenroller) [reply parlai/tasks/eli5/README.md:2]: Agreed

Comment (stephenroller) [reply parlai/tasks/eli5/README.md:2]: If you really want, you can like, do this:

````markdown
# ELI5

This is the ParlAI adaptation of the ELI5 dataset. It contains a lot of moving parts, most of which are adapted from the [original repo](https://github.com/facebookresearch/ELI5/).

The instructions for using it are as follows:
1. Step1
2. Step2

Below this point, you will find the **original README** taken from the ELI5 repo. It is left here only to provide additional context, and does NOT reflect installation instructions in ParlAI.

-----
Continued....
```

File: parlai/tasks/eli5/agents.py [hunk 20]
```diff
@@ -0,0 +1,72 @@
...
+            help='Whether to include supporting document knowledge',
+        )
+
+    def load_eli5(self, opt):
+        """
+        Load data based on data split.
+        """
+        dp = opt['datapath']
+        dt = opt['datatype'].split(':')[0]
+        eli_path = "eli5/processed_data/selected_15_1/explainlikeimfive_"
```
Comment (stephenroller) [inline parlai/tasks/eli5/agents.py:41]: This needs a a try/except path, that will raise an error when the user hasn't run the downloading instructions completely.

Comment (stephenroller) [reply parlai/tasks/eli5/agents.py:41]: And by try except path, I mean, we should loudly error and point the user to the docs.

File: parlai/tasks/eli5/README.md (hunk 2 above)
Comment (domrigoglioso) [reply parlai/tasks/eli5/README.md]: I changed the header section a little bit, but the rest is pretty much just the steps to get the dataset. Also, I left the citation section in there. Not sure if I should leave that or not

File: parlai/tasks/eli5/agents.py (hunk 20 above)
Comment (domrigoglioso) [reply parlai/tasks/eli5/agents.py:41]: Should throw a FileNotFoundError and tell them to follow the instructions now.

Comment (stephenroller) [reply parlai/tasks/eli5/agents.py:41]: That works for me.
